# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/cookiestore/CookieStore.cpp
# Commit: 4374d7fb7f40
# Full Hash: 4374d7fb7f40240e75f41519202eaf8127f323cc
# Author: Valentin Gosu <valentin.gosu@gmail.com>
# Date: 2025-03-29 09:19:43
# Regressor Bug: 1951750
# File Overlap Count: 1
# Description:
#   Bug 1951750 - Don't allow space in cookieStore names or values r=baku
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D241967
# ==============================================================================

diff -r bb99a3d0d1e7 -r 4374d7fb7f40 dom/cookiestore/CookieStore.cpp
--- a/dom/cookiestore/CookieStore.cpp	Fri Mar 28 13:48:31 2025 +0000
+++ b/dom/cookiestore/CookieStore.cpp	Fri Mar 28 13:48:31 2025 +0000
@@ -48,6 +48,9 @@
     if (*iter == 0x3B || *iter == 0x7F || (*iter <= 0x1F && *iter != 0x09)) {
       return false;
     }
+    if (*iter == 0x20) {
+      return false;
+    }
   }
   return true;
 }
